src/core/coap/coap.cpp

```diff
@@ -979,10 +1010,10 @@ uint32_t TxParameters::CalculateMaxTransmitWait(void) const
     return CalculateSpan(mMaxRetransmit + 1);
 }
 
-uint32_t TxParameters::CalculateSpan(uint32_t aMaxRetx) const
+uint32_t TxParameters::CalculateSpan(uint8_t aMaxRetx) const
 {
-    return static_cast<uint32_t>(mAckTimeout * ((1ULL << aMaxRetx) - 1) * mAckRandomFactorNumerator /
-                                 mAckRandomFactorDenominator);
+    return static_cast<uint32_t>(mAckTimeout * ((1U << aMaxRetx) - 1) / mAckRandomFactorDenominator *
```

Comment:
could incorrectly calculate to 0 if `mAckTimeout * ((1U << aMaxRetx) - 1)` < `mAckRandomFactorDenominator`?

Reply:
added verification of `mAckTimeout` is no less than 1.0.